pkg/models/plan_payments.go

```diff
@@ -0,0 +1,133 @@
+	CanParticipantsSelectBetweenPaymentMechanismsNote        *string                                                         `json:"canParticipantsSelectBetweenPaymentMechanismsNote" db:"can_participants_select_between_payment_mechanisms_note"`
+	AnticipatedPaymentFrequency                              anticipatedpaymentfrequencytype.AnticipatedPaymentFrequencyType `json:"anticipatedPaymentFrequency" db:"anticipated_payment_frequency"`
+	AnticipatedPaymentFrequencyOtherDescription              *string                                                         `json:"anticipatedPaymentFrequencyOtherDescription" db:"anticipated_payment_frequency_other_description"`
+	AnticipatedPaymentFrequencyNotes                         *string                                                         `json:"anticipatedPaymentFrequencyNotes" db:"anticipated_payment_frequency_notes"`
+
+	// Page 7
+	WillRecoverPayments                               *bool      `json:"willRecoverPayments" db:"will_recover_payments"`
+	WillRecoverPaymentsNotes                          *string    `json:"willRecoverPaymentsNotes" db:"will_recover_payments_notes"`
+	AnticipateReconcilingPaymentsRetrospectively      *bool      `json:"anticipateReconcilingPaymentsRetrospectively" db:"anticipate_reconciling_payments_retrospectively"`
+	AnticipateReconcilingPaymentsRetrospectivelyNotes *string    `json:"anticipateReconcilingPaymentsRetrospectivelyNotes" db:"anticipate_reconciling_payments_retrospectively_notes"`
+	PaymentStartDate                                  *time.Time `json:"paymentStartDate" db:"payment_start_date"`
+	PaymentStartDateNotes                             *string    `json:"paymentStartDateNotes" db:"payment_start_date_notes"`
+
+	// Meta
+	CreatedBy   string     `json:"createdBy" db:"created_by"`
+	CreatedDts  time.Time  `json:"createdDts" db:"created_dts"`
+	ModifiedBy  *string    `json:"modifiedBy" db:"modified_by"`
+	ModifiedDts *time.Time `json:"modifiedDts" db:"modified_dts"`
+	Status      TaskStatus `json:"status" db:"status"`
+}
+
+// GetModelTypeName provides the PlanPayments model type name
+func (p PlanPayments) GetModelTypeName() string {
+	return "Plan_Payments"
+}
+
+// GetID provides the PlanPayments ID
+func (p PlanPayments) GetID() uuid.UUID {
+	return p.ID
+}
+
+// GetPlanID provides the PlanPayments associated model plan ID
+func (p PlanPayments) GetPlanID() uuid.UUID {
+	return p.ModelPlanID
+}
+
+// GetCreatedBy provides the CreatedBy field
+func (p PlanPayments) GetCreatedBy() string {
+	return p.CreatedBy
+}
+
+// GetModifiedBy provides the ModifiedBy field
+func (p PlanPayments) GetModifiedBy() *string {
+	return p.ModifiedBy
+}
+
+// CalcStatus calculates the status of the Plan Payments and sets the Status field
+func (p PlanPayments) CalcStatus() error {
+	status, err := GenericallyCalculateStatus(p)
```

Comment:
I think it's good to use this, but the fields need the `statusWeight` tag of this should error out. EG `statusWeight:"1"`. When you pull in main, the updated method should now support pq.stringArrays as well as pointers.